### ISSUE TITLE
Problem unzipping some xlsx files

### ISSUE THREAD

**Issue by jgwconsulting** [rubyzip/rubyzip#105]
Throwing 'Zip end of central directory signature not found (Zip::ZipError)'

See these two issues from separate gems that use RubyZip:

https://github.com/Empact/roo/issues/80
https://github.com/gilt/rubyXL/issues/60

**Comment by slavarybalka** [comment]
getting this exception when trying to process some of the docx files
using require 'docx' 
on some files my script works fine and on the others it throws this exception

**Comment by zaeem** [comment]
@weshatheleopard  It's .xlsx file and I am still getting error on windows and mac. You can see for yourself. here is the link. https://www.dropbox.com/s/5tnb7h2mm9h47ef/long%2Breport-4-11-01-2016.zip?dl=0

**Comment by MattPorto** [comment]
Well, in 2019 I faced this bug, and for those who faced it too, here is how I fix this:

A error explanation (skip if you just want to fix it): 
First, this is a `.xlsx` file problem. In my case, if I open the `.xlsx` file on LibreOffice, first the LibreOffice points a format error in a specific place of the spreadsheet. Second if I try to save it with a different name I got a format/content error warning by LibreOffice.

The solution:
I'm using `roo` to handle `.xlsx` files, and the @duhast answer above makes me got it. All I did was add a rescue for this error and save the file again as a temporary `.csv` file (on my context I don't need to care what is the specific format of the file actually, I just need to read them and extract it's data). 

I have a [gist](https://gist.github.com/MattPorto/7ecbd1153756a5396dbfc96948cb0e97.js) exemplifying the snippet mentioned above.
